crates/payload/validator/src/lib.rs

```diff
@@ -116,9 +122,42 @@ impl ExecutionPayloadValidator {
 
         let cancun_active = self.is_cancun_active_at_timestamp(sealed_block.timestamp);
 
-        if !cancun_active && sealed_block.has_blob_transactions() {
-            // cancun not active but blob transactions present
-            return Err(PayloadError::PreCancunBlockWithBlobTransactions)
+        if !cancun_active {
+            if sealed_block.has_blob_transactions() {
+                // cancun not active but blob transactions present
+                return Err(PayloadError::PreCancunBlockWithBlobTransactions)
+            }
+            if sealed_block.header.blob_gas_used.is_some() {
+                // cancun not active but blob gas used present
+                return Err(PayloadError::PreCancunBlockWithBlobGasUsed)
+            }
+            if sealed_block.header.excess_blob_gas.is_some() {
+                // cancun not active but excess blob gas present
+                return Err(PayloadError::PreCancunBlockWithExcessBlobGas)
+            }
+            if cancun_fields.clone().into_inner().is_some() {
+                // cancun not active but cancun fields present
+                return Err(PayloadError::PreCancunWithCancunFields)
+            }
+        } else {
+            if sealed_block.header.blob_gas_used.is_none() {
```

Comment:
very pedantic nit, but I'd like to flip the if condition and the branches

Reply:
sure done here https://github.com/paradigmxyz/reth/pull/7993/commits/d1f697228c54cc303d25c2532d9118819493d410